cfripper/rules/GenericWildcardPrincipalRule.py

```diff
@@ -59,7 +59,7 @@ def check_for_wildcards(self, logical_id: str, resource: PolicyDocument):
             if account_id_match:
                 self.validate_account_id(logical_id, account_id_match.group(1))
 
-            if not self.resource_is_whitelisted(logical_id):
+            if not resource.Condition and not self.resource_is_whitelisted(logical_id):
```

Comment:
When was this removed? The original rule had a check for statement.condition in `check_principals`

Reply:
it was removed in the big update 😞

Reply:
👍 👍